lib/improver/weighted_blend.py

```diff
@@ -33,13 +33,208 @@
+        return result
+
+    @staticmethod
+    def blend_percentile_cube(cube, weights, coord, coord_dim, perc_coord):
+        """ Blend together percentile cube
+
+        Args:
+            cube : iris.cube.Cube
+                   Cube to blend across the coord.
+            weights: Optional list or np.array of weights
+                     or None (equivalent to equal weights)
+            coord : string
+                     The name of a coordinate dimension in the cube.
+            coord_dim : tuple
+                     The index of the coordinate dimension in the cube.
+            perc_coord : iris.cube.DimCoord
+                     The perecentile coordinate
+
+        Returns:
+            result : iris.cube.Cube
+                     containing the weighted percentile blend
+                     across the chosen coord
+        """
+
+        percentiles = np.array(perc_coord.points, dtype=float)
+        num = cube.data.shape[coord_dim[0]]
+        if weights is None:
+            weights = np.ones(num) / float(num)
+        PERCENTILE_BLEND = (Aggregator('percentile_blend',
+                            BlendingUtilities.blend_percentile_aggregate))
+        perc_dim, = cube.coord_dims(perc_coord.name())
+
+        result = cube.collapsed(coord,
+                                PERCENTILE_BLEND,
+                                arr_percent=percentiles,
+                                arr_weights=weights,
+                                perc_dim=perc_dim)
+        return result
+
+    @staticmethod
+    def blend_percentile_aggregate(data, axis,
+                                   arr_percent, arr_weights, perc_dim):
+        """ Blend percentile aggregate function
+
+        Args:
+            data : np.array
+                   Array containing the data to blend
+            axis : integer
+                   The index of the coordinate dimension in the cube.
+            arr_percent: np.array
+                     Array of percentile values e.g
+                     [0, 20.0, 50.0, 70.0, 100.0],
+                     same size as the percentile dimension of data.
+            arr_weights: np.array
+                     Array of weights, same size as the axis dimension of data.
+            perc_dim : integer
+                     The index of the perecentile coordinate
+            (Note percent and weights have special meaning in Aggregator
+             hence the rename.)
+
+        Returns:
+            result : np.array
+                     containing the weighted percentile blend data
+                     across the chosen coord
+        """
+        if axis < 0:
+            axis += data.ndim
+        # Firstly ensure axis coordinate and percentile coordinate
+        # are indexed as the first and second values in the data array
+
+        data = np.moveaxis(data, [perc_dim, axis], [1, 0])
+
+        # Determine the rest of the shape
+        shape = data.shape[2:]
+        result = None
+        if shape:
```

Comment:
This doesn't work, and I don't think this is the correct behaviour. I'll try and fix it.